p2p/discover/table_util_test.go

```diff
@@ -199,6 +199,8 @@ func checkNodesEqual(got, want []*enode.Node) error {
 				goto NotEqual
 			}
 		}
+	} else {
+		return fmt.Errorf("length dismatch: got %d nodes, want %d nodes", len(got), len(want))
```

Comment:
Alternative fix: just move the `return nil` up one line, into the if-clause. Then the non-equal case would just fall through to the code below

Reply:
@holiman Thanks for reviewing. What you suggest seems more elegant :)

But what more important is that test case `TestUDPv5_lookupE2E` is defective, `UDPv5.Lookup` actually could not find all expected nodes. So if we merge this bugfix, this test case will never pass.

Reply:
@holiman  I try to fix `TestUDPv5_lookupE2E`, but it seem hard for me now :)

Maybe you guys could fix it, then this PR could get merged.